### PR TITLE
feature/PATRO23-35 - add isFavourite flag

### PR DESCRIPTION
**Pull request by IzabelaPalubicka**
https://tracker.intive.com/jira/browse/PATRO23-35

### CODE REVIEW

File: src/modules/budget/application/Budget/Mappers/BudgetAggregateBudgetInfoMapper.cs
Comment (Kepeczsz) [inline src/modules/budget/application/Budget/Mappers/BudgetAggregateBudgetInfoMapper.cs:1]: typo "fourite"

File: src/modules/budget/contracts/Events/UserBudgetFavouriteUpdatedDomainEvent.cs
Comment (Kepeczsz) [inline src/modules/budget/contracts/Events/UserBudgetFavouriteUpdatedDomainEvent.cs:1]: NewIsFavourite doesn't feel right for me, maybe something like IsNewBudgetFavourite, also summary could be more detailed.

**Review by DawidBski — APPROVED** [review]

**Review by KlaudiaGajocha — APPROVED** [review]

**Review by Kepeczsz — APPROVED** [review]

**Review by LukaszWinowicz — APPROVED** [review]

File: src/modules/budget/application/EventHandlers/UserBudgetFavouriteUpdatedDomianEventHandler.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+using Intive.Patronage2023.Shared.Infrastructure.EventHandlers;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Intive.Patronage2023.Modules.Budget.Application.EventHandlers;
+
+/// <summary>
+/// UserBudget favourite updated domain event handler.
+/// </summary>
+[Lifetime(Lifetime = ServiceLifetime.Singleton)]
+public class UserBudgetFavouriteUpdatedDomianEventHandler : IDomainEventHandler<UserBudgetFavouriteUpdatedDomainEvent>
```
Comment (mateuszmuszynski) [inline src/modules/budget/application/EventHandlers/UserBudgetFavouriteUpdatedDomianEventHandler.cs:12]: typo, **Domian**EventHandler

File: src/modules/budget/contracts/Events/UserBudgetFavouriteUpdatedDomainEvent.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	/// <param name="isNewBudgetFavourite">New IsFavourite value.</param>
+	public UserBudgetFavouriteUpdatedDomainEvent(bool isNewBudgetFavourite)
+	{
+		this.IsNewBudgetFavourite = isNewBudgetFavourite;
+	}
+
+	/// <summary>
+	/// New UserBudget isFavourite value.
+	/// </summary>
+	public bool IsNewBudgetFavourite { get; set; }
```
Comment (mateuszmuszynski) [inline src/modules/budget/contracts/Events/UserBudgetFavouriteUpdatedDomainEvent.cs:22]: Imo "New" part is unnecessary.

File: src/modules/budget/api/BudgetModule.cs [hunk 3]
```diff
@@ -6,14 +6,15 @@
 using Intive.Patronage2023.Modules.Budget.Application.Budget.EditingBudget;
```
Comment (mateuszmuszynski) [inline src/modules/budget/api/BudgetModule.cs:6]: Add tests (applies to all changes, not this file)

**Review by mateuszmuszynski — CHANGES_REQUESTED** [review]

File: src/modules/budget/api/BudgetModule.cs (hunk 3 above)
Comment (IzabelaPalubicka) [reply src/modules/budget/api/BudgetModule.cs:6]: We discussed that tests will be added in another task https://tracker.intive.com/jira/browse/PATRO23-346

**Review by mateuszgiedzinski — APPROVED** [review]

**Review by mateuszmuszynski — APPROVED** [review]

**Pull request merged**